Automattic/jetpack
# Blocks: add a new Cookie Consent Block

## Proposed changes:

Replay of #29191 in the Jetpack plugin.

This adds a new Beta block, to be used in block themes, to display a cookie consent banner on the site.

<img width="784" alt="image" src="https://user-images.githubusercontent.com/426388/221796997-a38b425e-2e42-4b80-947e-8114c5bec7db.png">

### Other information:


## Jetpack product discussion

* pdDR7T-Ax-p2
* pdDR7T-Cr-p2

## Does this pull request change what data or activity we track or use?

* Yes, it adds a new cookie (`eucookielaw`) when you click to accept the banner. That cookie is already in use by the Cookies & Consents Banner widget, but that should still be documented.

## Testing instructions:

1. Start with a site that's connected to WordPress.com and where you have enabled Beta blocks (`add_filter( 'jetpack_blocks_variation', function () { return 'beta'; } );`).
    * On WordPress.com Simple, you do not need the filter. You will see the block when you are logged in.
2. Go to Appearance > Themes and enable a block theme; Twenty Twenty Three will work.
3. Go to Appearance > Site Editor
4. You should now be able to add and customize the block. 
    * Check that you can edit contents and change the settings in the block sidebar.
    * Check that your changes can be saved.
    * Check that your changes are visible on the frontend.
    * When dismissing the banner in the frontend, you will see the `eucookielaw` cookie be added in your browser. The banner should then be gone.
    * When refreshing the page, the banner should still be gone.
